core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -266,8 +267,18 @@ private[spark] class TaskSchedulerImpl(
               assert(availableCpus(i) >= 0)
               launchedTask = true
             }
+          } catch {
+            case e: TaskNotSerializableException => {
+              return
```

Comment:
This could probably use a descriptive comment to explain that the error is already logged elsewhere.  On the other hand, I suppose that it doesn't hurt to add a redundant logging statement here, too.